mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividual/NewIndividualCollectionSheetFragment.kt

```diff
@@ -4,273 +4,111 @@ import android.os.Bundle
-    private var officeId: Int? = 0
-    private var staffId: Int? = 0
-    private val requestCode = 1
-    private var success = true
-    private var actualDisbursementDate: String? = null
-    private var transactionDate: String? = null
-
-    private var selectedRepaymentDate: Instant = Instant.now()
-    private val datePickerDialog by lazy {
-        getDatePickerDialog(selectedRepaymentDate, DatePickerConstrainType.ONLY_FUTURE_DAYS) {
-            val formattedDate = SimpleDateFormat("dd MMMM yyyy", Locale.getDefault()).format(it)
-            selectedRepaymentDate = Instant.ofEpochMilli(it)
-            binding.repaymentDateFieldContainer.editText?.setText(formattedDate)
-        }
-    }
+    private lateinit var repaymentDate: String
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         if (savedInstanceState != null) {
             sheet =
                 savedInstanceState[Constants.EXTRA_COLLECTION_INDIVIDUAL] as IndividualCollectionSheet
         }
-        setHasOptionsMenu(true)
     }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        binding = FragmentNewCollectionSheetBinding.inflate(inflater, container, false)
-        setToolbarTitle(getStringMessage(R.string.individual_collection_sheet))
-        viewModel = ViewModelProvider(this)[NewIndividualCollectionSheetViewModel::class.java]
-        setUpUi()
-
-        viewModel.newIndividualCollectionSheetUiState.observe(viewLifecycleOwner) {
-            when (it) {
-                is NewIndividualCollectionSheetUiState.SetOfficeSpinner -> {
-                    showProgressbar(false)
-                    setOfficeSpinner(it.officeList)
-                }
-
-                is NewIndividualCollectionSheetUiState.SetStaffSpinner -> {
-                    showProgressbar(false)
-                    setStaffSpinner(it.staffList)
-                }
-
-                is NewIndividualCollectionSheetUiState.ShowError -> {
-                    showProgressbar(false)
-                    showError(it.errorMessage)
-                }
-
-                is NewIndividualCollectionSheetUiState.ShowNoSheetFound -> {
-                    showProgressbar(false)
-                    showNoSheetFound()
-                }
-
-                is NewIndividualCollectionSheetUiState.ShowProgressbar -> showProgressbar(true)
-                is NewIndividualCollectionSheetUiState.ShowSheet -> {
-                    showProgressbar(false)
-                    showSheet(it.individualCollectionSheet)
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(
+                ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed
+            )
+            setContent {
+                val viewModel: NewIndividualCollectionSheetViewModel = hiltViewModel()
```

Comment:
@PratyushSingh07 can you help @Aditya-gupta99 about injecting viewmodel in compose function

Reply:
I discussed with Pratyush, but we should not inject viewmodel as a parameter in compose. instead, we should only give state because it will cause problems when doing compose tests.

Reply:
I have asked him to make those changes